### ISSUE TITLE
Better error message when argument list in call expression is malformed

### ISSUE THREAD

**Issue by affo** [influxdata/flux#1235]
This works fine:

```
from(bucket: "Default")
	|> range(start: -5m)
	|> filter(fn: (r) => r._measurement == "cpu" and r._field == "usage_system" and r.cpu == "cpu-total")
	|> map(fn: (r) => ({_v:  r._value, index: 1}))
```

This one not:

```
from(bucket: "Default")
	|> range(start: -5m)
	|> filter(fn: (r) => r._measurement == "cpu" and r._field == "usage_system" and r.cpu == "cpu-total")
	|> map(fn: (r) => ({_v:  float(r._value), index: 1}))
```

```
Error: Failed to execute query: type error 4:36-4:42: undefined identifier "_value".
```

**Comment by affo** [comment]
This is actually a better error message problem, and it should be fixed once https://github.com/influxdata/flux/pull/1175 gets merged.

The correct script is:

```
from(bucket: "Default")
	|> range(start: -5m)
	|> filter(fn: (r) => r._measurement == "cpu" and r._field == "usage_system" and r.cpu == "cpu-total")
	|> map(fn: (r) => ({_v:  float(v: r._value), index: 1}))
```

Note `float(v: ...)`.

**Comment by russorat** [comment]
@affo is this fixed now that #1175 is merged?

**Comment by affo** [comment]
I'd say no, now it returns:

```
Error: unsupported key <nil>
```

I'll work on this, it should be pretty straight-forward.
